### PR TITLE
Fix "View on GitHub" links

### LINKED ISSUE #1650: Checkbox, Radio, and Switch Example Github Links are broken
### Bug report
- __Package version(s)__: core 1.29.0
- __Browser and OS versions__: Mac OSX High Sierra (10.13 (17A365)), Chrome (Version 61.0.3163.100 (Official Build) (64-bit))

Checkbox, Switch, and Radio Github example links are broken (lead to 404 Not Found).

I suspect this is due to the way in which Blueprint currently generates `sourceUrl`. It seems that `sourceUrl` is generated for all examples in the file `reactExamples.ts` through the following code:

```ts
function addPackageExamples(
    packageName: string,
    packageExamples: { [name: string]: React.ComponentClass<IBaseExampleProps> },
) {
    for (const exampleName of Object.keys(packageExamples)) {
        const example = packageExamples[exampleName];
        **const fileName = exampleName.charAt(0).toLowerCase() + exampleName.slice(1) + ".tsx";**
        reactExamples[exampleName] = {
            render: props => React.createElement(example, { ...props, themeName: getTheme() }),
            sourceUrl: [SRC_HREF_BASE, packageName, "examples", fileName].join("/"),
        };
    }
}
```

The problem is that both the Checkbox, Switch, and Radio examples are exported from a larger file called `controlExamples.tsx`, and do not each have their own files. This means the above code is generating `sourceUrl`s for files that do not exist.

Some ideas for a fix are by separating Radio/Switch/Checkbox into their own files, or by changing `reactExample.tsx` to account for this inconsistency (is there some way to access `import` filepaths?).

Tiny bug, but would love an opportunity to start contributing to this fantastic project in whatever ways possible!

#### Steps to reproduce

1. Navigate to BlueprintJS (either locally or directly to the [official BlueprintJS docs](http://blueprintjs.com/docs/#core/components/forms/radio))
2. Navigate to either the Form Controls > Radio Button page, OR the Form Controls > Checkbox page
2. Click on "View on Github"

#### Actual behavior
Github returns a 404 page not found error.

#### Expected behavior

Should return a Github page containing the example source code.

### PR DESCRIPTION
**Pull request by zfranklyn**
#### Fixes #1650 

Fixed "View On Github" links by separating `controlExamples.tsx` into individual files: `radioExample.tsx`, `checkboxExample.tsx`, and `switchExample.tsx`. This should work only when actually merged in, or else the new Github URLs will be invalid.

#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [x] Include tests (there were no tests for examples)
- [x] Update documentation (no major changes to merit documentation edits)

#### Changes proposed in this pull request:

Split `controlExamples.tsx` into individual files: `radioExample.tsx`, `checkboxExample.tsx`, and `switchExample.tsx`. 

#### Reviewers should focus on:

My first time contributing, so please let me know if there's anything wrong at all (especially with merging or CI).

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @zfranklyn! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/core/examples/radioExample.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import * as React from "react";
+
+import { Radio, RadioGroup } from "@blueprintjs/core";
+import { BaseExample, handleStringChange } from "@blueprintjs/docs";
+
+export interface IRadioExampleState {
+    radioValue?: string;
+}
+
+// tslint:disable-next-line:max-classes-per-file
```
Comment (giladgray) [inline packages/core/examples/radioExample.tsx]: should be safe to enable this rule now (remove the line). was only necessary cuz we had multiple things in one file (more proof this PR is correct).

File: packages/core/examples/switchExample.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as React from "react";
+
+import { Classes, Switch } from "@blueprintjs/core";
+import { BaseExample } from "@blueprintjs/docs";
+
+// tslint:disable-next-line:max-classes-per-file
```
Comment (giladgray) [inline packages/core/examples/switchExample.tsx]: same

**Comment by zfranklyn** [comment]
<h3>removed unnecessary tslint disabling</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/zfranklyn/blueprint/4/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/zfranklyn/blueprint/4/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/zfranklyn/blueprint/4/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: packages/core/examples/checkboxExample.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as React from "react";
+
```
Comment (cmslewis) [inline packages/core/examples/checkboxExample.tsx]: nit: we typically chunk imports from other `@blueprintjs` packages and external dependencies together now:

```
import * as React from "react";
import { Checkbox, Classes } from "@blueprintjs/core";
import { BaseExample } from "@blueprintjs/docs";

// import * from "../some/relative/path/in/this/package
// import * from "../another/relative/path/in/this/package
```

**Pull request merged**